### DISCUSSION TITLE
(solved)index state NONE after creating index

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#16037] [Q&A and General discussion]
Hi, I use cpp sdk to build an index, but when I build it, time cost of creating an index is 1ms. So I print the index state of this index, the state of this index is NONE.
-----------------------------------------------
my code is like below:

    std::string index_name = "IVFPQ";
    int64_t index_id = 0;
    int alpha = 2;
    int32_t nlist = alpha * std::sqrt(mid_vec.size() );
    const std::unordered_map<std::string, std::string> params = {
        {"nlist", std::to_string(nlist)},
        {"m", "8"},
    };
    milvus::IndexDesc index_desc(field_embedding_name, index_name, index_id, params);
    const auto progress_monitor = ::milvus::ProgressMonitor::NoWait();
    status = client->CreateIndex(collection_name, index_desc, progress_monitor);

**Comment by xiaojuan8** [comment]
Hi, I found sdk-cpp has updated, IndexDesc class has changed. I modified my client code to new version style. It turns right. I successfully created my index. yeh

**Comment by yhmo** [comment]
That is a good news.
If you found any other problems/bugs, please report to us.
